web/src/main/java/org/springframework/security/web/server/header/DelegatingServerHttpHeadersWriter.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2002-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.web.server.header;
+
+import reactor.core.publisher.Mono;
+
+import org.springframework.security.web.server.util.matcher.ServerWebExchangeMatcher;
+import org.springframework.util.Assert;
+import org.springframework.web.server.ServerWebExchange;
+
+/**
+ * Delegates to a provided {@link ServerHttpHeadersWriter} when
+ * {@link ServerWebExchangeMatcher#matches(ServerWebExchange)} returns a match.
+ *
+ * @author David Herberth
+ * @since 5.7
+ */
+public final class DelegatingServerHttpHeadersWriter implements ServerHttpHeadersWriter {
+
+	private final ServerWebExchangeMatcher webExchangeMatcher;
+
+	private final ServerHttpHeadersWriter delegateHeadersWriter;
```

Comment:
What do you think about using `ServerWebExchangeMatcherEntry<ServerHttpHeadersWriter>` class here? The constructor could benefit from it as well, receiving just one parameter.
```suggestion
	private final ServerWebExchangeMatcherEntry<ServerHttpHeadersWriter> delegate;
```

Reply:
Thanks I didn't know this class existed, I think having it as an overload for the constructor is good/fine, but I would like to keep the current constructor, simply because it makes the calling code more readable:

```java
new ServerWebExchangeDelegatingServerHttpHeadersWriter(
    ServerWebExchangeMatchers.pathMatchers("/**"), 
    StaticServerHttpHeadersWriter.builder().headers("foo", "bar").build()
);

// or this:
new ServerWebExchangeDelegatingServerHttpHeadersWriter(new ServerWebExchangeMatcherEntry(
    ServerWebExchangeMatchers.pathMatchers("/**"), 
    StaticServerHttpHeadersWriter.builder().headers("foo", "bar").build()
));
```